From e6f3da53583853731a399c62a6271e06399e4a64 Mon Sep 17 00:00:00 2001
From: Zhiwei Yang <quic_zhiwyang@quicinc.com>
Date: Wed, 10 Apr 2024 10:07:21 +0800
Subject: [PATCH] qcacld-3.0: Use cfg80211_register_netdevice() for kernel 4.19.124+

For Kernel 4.19.124+ in order to be in line with cfg80211 changes
in upstream code, use cfg80211_register_netdevice() and
cfg80211_unregister_netdevice() when request to register device is
coming from add virtual interface.

Signed-off-by: Zhiwei Yang <quic_zhiwyang@quicinc.com>
---
 core/hdd/src/wlan_hdd_main.c       | 4 ++--
 core/hdd/src/wlan_hdd_regulatory.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/hdd/src/wlan_hdd_main.c b/core/hdd/src/wlan_hdd_main.c
index 0be3ed1328..4b2056bfbd 100644
--- a/core/hdd/src/wlan_hdd_main.c
+++ b/core/hdd/src/wlan_hdd_main.c
@@ -7448,7 +7448,7 @@ free_net_dev:
 	return NULL;
 }
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0) || \
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 124) || \
 	(defined CFG80211_CHANGE_NETDEV_REGISTRATION_SEMANTICS))
 static int
 hdd_register_netdevice(struct hdd_adapter *adapter, struct net_device *dev,
@@ -8695,7 +8695,7 @@ static void hdd_sta_destroy_ctx_all(struct hdd_context *hdd_ctx)
 	}
 }
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0) || \
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 124) || \
 	(defined CFG80211_CHANGE_NETDEV_REGISTRATION_SEMANTICS))
 static void
 hdd_unregister_netdevice(struct hdd_adapter *adapter, struct net_device *dev)
diff --git a/core/hdd/src/wlan_hdd_regulatory.c b/core/hdd/src/wlan_hdd_regulatory.c
index 98a971bc44..aad4bd4e6d 100644
--- a/core/hdd/src/wlan_hdd_regulatory.c
+++ b/core/hdd/src/wlan_hdd_regulatory.c
@@ -1475,7 +1475,7 @@ static inline void hdd_set_dfs_pri_multiplier(struct hdd_context *hdd_ctx,
 }
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)) || \
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 124)) || \
 	defined(CFG80211_WIPHY_REGD_NO_RTNL_SUPPORT)
 static int
 hdd_regulatory_set_wiphy_regd_sync(struct wiphy *wiphy,
-- 
2.7.4

